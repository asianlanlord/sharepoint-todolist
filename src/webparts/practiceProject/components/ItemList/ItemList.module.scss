@import '~office-ui-fabric-react/dist/sass/References.scss';

.flexContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    text-align: center;
    justify-content: center;
    width: 100%;
}

.pageHeader {
    font-size: 2em;
    font-weight: bold;
    text-align: center;
}

.title{
    flex: 2;
    padding: 5px;
}

.description{
    flex: 5;
    padding: 5px;
    background-color: lightblue;
}

.priority{
    flex: 2;
    padding: 5px;
}

.dueDate{
    flex: 2;
    padding: 5px;
    background-color: lightblue;
}

.buttonContainer{
    flex: 3;
    padding: 5px;
}
