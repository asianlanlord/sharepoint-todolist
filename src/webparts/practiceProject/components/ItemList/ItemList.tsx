import * as React from 'react';
import styles from './ItemList.module.scss';
import { useObserver } from 'mobx-react-lite';
import store from '../../ToDoListStore';
import CustomButton from '../CustomButton/CustomButton';
import PriorityHighlighter from '../PriorityHighlighter/PriorityHighlighter';

export interface ItemListProps {
  webUrl?: any;
  listID?: any;
}

function ItemList(props: ItemListProps) {
  return useObserver(() => (
    <div id="itemList" className={styles.itemList}>
      <p className={styles.pageHeader}>To Do List</p>

      <div className={styles.addButton}>
        <CustomButton buttonText={"Add task"} listID={props.listID} webUrl={props.webUrl} />
      </div>

      <div className={styles.flexContainer}>
        <div className={styles.titleHeader}>  Task </div>
        <div className={styles.descriptionHeader}>  Task Description </div>
        <div className={styles.priorityHeader}> Priority </div>
        <div className={styles.dueDateHeader}>  Due Date </div>
        <div className={styles.buttonHeader}> Actions </div>
      </div>
      {store.list.length > 0 ? store.list.map((listItems) => {
        return (
          <div className={styles.flexContainer}>
            <div className={styles.title}>  {listItems.Title} </div>
            <div className={styles.description}>  {listItems.Description} </div>
            <div className={styles.priority}> <PriorityHighlighter priority={listItems.Priority} /> </div>
            <div className={styles.dueDate}>  {listItems.DueDate.slice(0, 10)} </div>
            <div className={styles.buttonContainer}>

              <CustomButton buttonText={"Remind"}
                listID={props.listID}
                webUrl={props.webUrl}
                itemID={listItems.ID}
                itemTitle={listItems.Title}
                itemDesc={listItems.Description}
                itemDue={listItems.DueDate.slice(0, 10)}
                itemPrio={listItems.Priority}
              />

              <CustomButton buttonText={"Edit"}
                listID={props.listID}
                webUrl={props.webUrl}
                itemID={listItems.ID}
                itemTitle={listItems.Title}
                itemDesc={listItems.Description}
                itemDue={listItems.DueDate.slice(0, 10)}   
                itemPrio={listItems.Priority}
              />

              <CustomButton buttonText={"Delete"}
                listID={props.listID}
                webUrl={props.webUrl}
                itemID={listItems.ID}
                itemTitle={listItems.Title}
                itemDesc={listItems.Description}
                itemDue={listItems.DueDate.slice(0, 10)}
                itemPrio={listItems.Priority}
              />
            </div>

          </div>
        );
      }
      ) : ""}
    </div>
  ));
}

export default ItemList;