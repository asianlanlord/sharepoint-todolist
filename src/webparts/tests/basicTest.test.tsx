import renderer from 'react-test-renderer';
import ItemList from '../practiceProject/components/ItemList/ItemList';
import React from 'react';

import { configure, mount, ReactWrapper } from 'enzyme';
const Adapter = require('enzyme-adapter-react-16')


configure({ adapter: new Adapter() });




it('Base Item list renders correctly', () => {
    const tree = renderer
      .create(<ItemList />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
