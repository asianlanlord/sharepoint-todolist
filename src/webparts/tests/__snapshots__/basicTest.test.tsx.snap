// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
     To Do List 
  </p>
  <div>
    <button
      className="ms-Button ms-Button--default root-85"
      data-is-focusable={true}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      <span
        className="ms-Button-flexContainer flexContainer-86"
        data-automationid="splitbuttonprimary"
      >
         
        Add new task
         
      </span>
    </button>
    
  </div>
  <div>
    <div>
        Task 
    </div>
    <div>
        Task Description 
    </div>
    <div>
       Priority 
    </div>
    <div>
        Due Date 
    </div>
    <div>
       Actions 
    </div>
  </div>
  
</div>
`;
